src/components/Dropdown/Dropdown.tsx

```diff
@@ -308,6 +352,96 @@ const Dropdown = ({
 
   const tagContainerItemProps = valueItemTagProps.containerProps || {};
 
+  // effect to determine if user is scrolling up or down
+  useEffect(() => {
+    const threshold = 0;
+    let lastScrollY = window.pageYOffset;
+    let ticking = false;
+
+    const updateScrollDir = () => {
+      const scrollY = window.pageYOffset;
+
+      if (Math.abs(scrollY - lastScrollY) < threshold) {
+        ticking = false;
+        return;
+      }
+      setIsScrollingDown(scrollY > lastScrollY);
+      lastScrollY = scrollY > 0 ? scrollY : 0;
+      ticking = false;
+    };
+
+    const onScroll = () => {
+      if (!ticking) {
+        window.requestAnimationFrame(updateScrollDir);
+        ticking = true;
+      }
+    };
+    window.addEventListener('scroll', onScroll);
+
+    return () => window.removeEventListener('scroll', onScroll);
+  }, []);
+
+  const buildThresholdArray = () => Array.from(Array(100).keys(), i => i / 100);
+  const intersectOptions = useMemo(() => {
+    return {
+      root: null,
+      rootMargin: '0px',
+      threshold: buildThresholdArray(),
+    };
+  }, []);
+
+  const intersectionCallback = useCallback(
+    (entries: IntersectionObserverEntry[]) => {
+      if (isAware) {
+        if (entries.length === 1) {
+          const [entry] = entries;
+          // swap the dropdown to open downward if its hitting the top
+          if (
+            entry.intersectionRatio < 0.95 &&
+            entry.target === optionsContainerInternalRef.current
+          ) {
+            if (isScrollingDown) {
+              if (!isOpenedBelow && entry.intersectionRatio < prevIntersectionRatio) {
+                setIsOpenedBelow(true);
+              }
+            }
+          }
+          setPrevIntersectionRatio(entry.intersectionRatio);
+        } else if (entries.length === 2) {
+          const [dropdown, invisibleDrop] = entries;
+          // flip the view if the other direction is more visible in viewport
+          if (invisibleDrop.intersectionRatio > dropdown.intersectionRatio) {
+            setIsOpenedBelow(drop => !drop);
+          }
+        }
+        setIsHidden(false);
+      } else {
+        setIsHidden(false);
+      }
+    },
+    [isOpenedBelow, isScrollingDown, prevIntersectionRatio, isAware],
+  );
+
+  const intersectObserver = useMemo(() => {
+    const observer = new IntersectionObserver(intersectionCallback, intersectOptions);
+    return observer;
+  }, [intersectOptions, intersectionCallback]);
```

Comment:
Prettier would make this look nicer than I have it here
```suggestion
  const intersectObserver = useMemo(() => new IntersectionObserver(intersectionCallback, intersectOptions), [intersectOptions, intersectionCallback]);
```